openedx/features/qverse_features/registration/validators.py

```diff
@@ -0,0 +1,32 @@
+"""
+Validation utils for qverse registration application.
+"""
+from django.core.exceptions import ValidationError
+
+
+def validate_file_extension(value):
+    """
+    Validates that file is a valid csv file by checking its extension.
+    """
+    if not value.name.endswith('.csv'):
+        raise ValidationError(u'Please upload a valid csv file.')
+
+
+def validate_current_level(value):
+    """
+    Validates the provided current level.
+    """
+    # to avoid circular import
+    from .models import MAX_LEVEL_CHOICES
+    if not 1 <= value <= MAX_LEVEL_CHOICES:
+        raise ValidationError(u'Please provide valid current level.')
+
+
+def validate_programme_choices(value):
+    """
+    Validates the provided programme choice.
+    """
+    # to avoid circular import
+    from .models import MAX_PROGRAMME_CHOICES
+    if not 1 <= value <= MAX_PROGRAMME_CHOICES:
```

Comment:
```suggestion
    from .models import PROGRAMME_CHOICES
    if value not in [choice[0] for choice in PROGRAMME_CHOICES]:
```
this way we will be independent of providing range, which means in future if we want to remove an option we don't have to refactor it here.

Reply:
I have tried all of these fancy things already. Actually I think this generator has already been consumed by `UserProfile` model itself so what we get is an empty list if we use your method
we get this
![image](https://user-images.githubusercontent.com/42294172/76949730-d61d2a80-692a-11ea-999e-6412182fbf22.png)
 
Working for one but still getting an empty list for `LEVEL_CHOICES`
So for consistency, it's better to keep the code as it is.
If you have any other method, suggest to me.

Reply:
can you share the value of level_choices? I don't know its value?

Reply:
it's given in model.py file.

```
MAX_LEVEL_CHOICES = 8
LEVEL_CHOICES = ((i, ugettext_noop('Year ' + str(i))) for i in range(1, MAX_LEVEL_CHOICES))
```